src/lib/ProductExporter.ts

```diff
@@ -186,6 +197,27 @@ export class ProductExporter {
             }
         }
     }
+    private retrieveReferenceChargesIds(chargesIds: Array<String>){
+        let chargeReferenceIds = new Array<any>();
+        chargesIds.filter(chargeId => chargeId['enxCPQ__Charge_Reference__r'])
+                  .forEach(chargeId => {chargeReferenceIds.push(chargeId['enxCPQ__Charge_Reference__r'])});
+        return chargeReferenceIds;
+    }
+
+    private async retrieveSecondaryProducts(conn: Connection, productList: Set<string>){
+        let secondaryProductNames =  new Set<string>();
+        let secondaryProducts = await Queries.querySecondaryProducts(conn, productList);
+        secondaryProducts.forEach( secondaryProduct => {secondaryProductNames.add(secondaryProduct['enxCPQ__Secondary_Product__r']['Name'])});
```

Comment:
Same here (map vs add) and maybe some other formatting? I mean, splitting into multiple lines.